### PR TITLE
Feature azure data explorer

### PR DESCRIPTION
**Pull request by gupta1**
This PR fixes #
This is a draft PR as suggested by @aaomidi so that we have a check on whether its passing all the required tests which we will need when its merged to main branch.

### CODE REVIEW

File: src/vs/loader.js [hunk 1]
```diff
@@ -22,532 +22,532 @@ var _amdLoaderGlobal = this;
...
-            }
-            if (typeof process !== 'undefined') {
-                return (process.platform === 'win32');
-            }
-            return false;
-        };
-        return Environment;
-    }());
-    AMDLoader.Environment = Environment;
+	AMDLoader.global = _amdLoaderGlobal;
```
Comment (aaomidi) [inline src/vs/loader.js:25]: Why was this changed?

File: extensions/kusto/config.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+	"downloadUrl": "https://github.com/Microsoft/sqltoolsservice/releases/download/v{#version#}/microsoft.sqltools.servicelayer-{#fileName#}",
```
Comment (kburtram) [inline extensions/kusto/config.json:2]: I wonder if it's better to share the same Tools Service binaries the mssql extension downloads to avoid having duplicate binaries on the machine.  I think this is what azure resource provider and credentials already do

**Comment by aaomidi** [comment]
So after talking to @gupta1 about this I recommended that they abandon this branch and remake a new one with only the relevant changes. 

A squash and merge caused this weird history syncing issue to come up, and it's pretty hard to verify that the changes are actually 0-diff from what main is.

File: src/vs/loader.js (hunk 1 above)
Comment (kburtram) [reply src/vs/loader.js:25]: Please revert changes to this file

File: src/vs/base/browser/ui/menu/menu.ts [hunk 3]
```diff
@@ -216,6 +216,15 @@ export class Menu extends ActionBar {
 			return true;
 		});
 
+		actions = actions.filter(a => {
```
Comment (kburtram) [inline src/vs/base/browser/ui/menu/menu.ts:219]: all changes to `vs` source files need to have a `// {{SQL CARBON EDIT}}` tag so we can track patches to vscode source.  I don't understand why this change is needed though.  Could you verify this isn't a merge issue and revert if not needed or add a comment if actually needed?

Comment (gupta1) [reply src/vs/base/browser/ui/menu/menu.ts:219]: @kburtram-I will take a look at it
